interpreter/cling/lib/Interpreter/IncrementalJIT.cpp

```diff
@@ -577,6 +581,18 @@ IncrementalJIT::IncrementalJIT(
 
   char LinkerPrefix = this->m_TM->createDataLayout().getGlobalPrefix();
 
+#if defined(__linux__) && defined(__GLIBC__)
+  // Inject lstat to pass test/DynamicLibraryManager/cached_realpath.C, it may
+  // be in libc_nonshared.a where symbols are not found automatically. See also
+  // upstream issue https://github.com/llvm/llvm-project/issues/61289.
+  SymbolMap LibcNonsharedSymbols;
+  LibcNonsharedSymbols[Jit->mangleAndIntern("lstat")] = JITEvaluatedSymbol(
```

Comment:
Can we bring all of the workarounds from [here](https://github.com/llvm/llvm-project/blob/3c6ed559e5274307995586c1499a2c8e4e0276a0/llvm/lib/ExecutionEngine/RuntimeDyld/RTDyldMemoryManager.cpp#L235)? I bet we will have bugs reports on the other functions at the moment we forget how to fix them :)

Reply:
I'd rather not since it's really a hack. This should be fixed properly upstream...

Reply:
I learn that would take time but I think we should update the related llvm issue in that case and keep it up to date.